config/default-spinnaker-local.yml

```diff
@@ -127,9 +127,9 @@ providers:
       name: my-docker-registry
       address: ${SPINNAKER_DOCKER_REGISTRY:https://index.docker.io/}
       repository: ${SPINNAKER_DOCKER_REPOSITORY:library/nginx}
-      username: ${SPINNAKER_DOCKER_USERNAME}
+      username: ${SPINNAKER_DOCKER_USERNAME:}
       # A path to a plain text file containing the user's password
-      passwordFile: ${SPINNAKER_DOCKER_PASSWORD_FILE}
+      passwordFile: ${SPINNAKER_DOCKER_PASSWORD_FILE:}
```

Comment:
Let's see if I understand your scenario correctly. I left clouddriver.yml and spinnaker.yml with the colons like in this pull request. I then added my username and password to my spinnaker-local.yml file like this:
> username: ${SPINNAKER_DOCKER_USERNAME:erijiz}
> password: examplePassword
> passwordFile: ${SPINNAKER_DOCKER_PASSWORD_FILE:}

That worked and I was able to authenticate to a private repo. My resolvedEnv looked like this:

> "providers.dockerRegistry.primaryCredentials.password": "******",
> "providers.dockerRegistry.primaryCredentials.passwordFile": "",
> "providers.dockerRegistry.primaryCredentials.repository": "erijiz/skuservice",
> "providers.dockerRegistry.primaryCredentials.username": "erijiz",

The resolvedEnv displays an empty string for passwordFile, which is better than getting the exception I listed above.